Base.rte/Devices/Tools/Constructor/Constructor.lua

```diff
@@ -52,9 +52,9 @@ function ConstructorFloodFill(x, y, startnum, maxnum, array, realposition, reals
 		end
 	end
 end
-
-function ConstructorSnapPos(checkPos)
-	return Vector(math.floor((checkPos.X - 12)/24) * 24 + 12, math.floor((checkPos.Y - 12)/24) * 24 + 12);
+--TODO: Figure out how to snap different sizes properly
+function ConstructorSnapPos(checkPos, blockSize)
+	return Vector(math.floor((checkPos.X - blockSize/2)/blockSize) * blockSize + blockSize/2, math.floor((checkPos.Y - blockSize/2)/blockSize) * blockSize + blockSize/2);
```

Comment:
There's SceneMan::SnapToPosition or something like that. Not sure if it's exposed to lua but it might be useful for ya.